src/dimension_api/__init__.py

```diff
@@ -0,0 +1,13 @@
+"""Dimension API."""
+
+from typing import Protocol, Self, runtime_checkable
+
+__version__ = "0.0.1.dev0"
+__all__ = ["Dimension"]
```

Comment:
I usually export `__version__` as well 🤷🏻 

```suggestion
__all__ = ["__version__", "Dimension"]
```